### ISSUE TITLE
no-unused-prop-types triggered when used in nextProps of UNSAFE_componentWillReceiveProps

### ISSUE THREAD

**Issue by KisaneNeko** [yannickcr/eslint-plugin-react#2363]
I'm using flow typings and I noticed this error after running script prefixing deprecated lifecycle methods. Basic example:

```
type Props = {
  dataDirty: boolean
};

class RandomComponent extends Component<Props> {
 UNSAFE_componentWillReceiveProps(nextProps: Props) {
    if (nextProps.dataDirty) {
      doSomething();
   }
 }
}
```

this triggers `'dataDirty' PropType is defined but prop is never used`. When UNSAFE_ prefix is removed the error is gone.

edit: it might be worth to mention, when searching I found this old issue:
https://github.com/yannickcr/eslint-plugin-react/issues/884
But in my case destructuring the nextProps didn't fix the error

**Comment by KisaneNeko** [comment]
Issue seems to be gone after updating eslint
